### PR TITLE
Configurable date range filter for entity lists

### PR DESCRIPTION
**Pull request by christophscheuing**
Date range filters can now be configured via the `config-fix.ts`, as described in #1448.

Configuration via UI, as described in #1383 is also implemented.

### CODE REVIEW

File: src/app/core/entity-components/entity-list/filter-generator.service.ts [hunk 1]
```diff
@@ -107,6 +111,89 @@ export class FilterGeneratorService {
     }
   }
 
+  private createDateRangeFilterOptions<T extends Entity>(
+    filterConfig: DateRangeFilterConfig
+  ): FilterSelectionOption<T>[] {
+    if (filterConfig.startingDayOfWeek) {
+      let configStartingDayOfWeek;
+      switch (filterConfig.startingDayOfWeek.toLowerCase()) {
```
Comment (TheSlimvReal) [inline src/app/core/entity-components/entity-list/filter-generator.service.ts:119]: lets use a enum for this, so we don't need a switch-case here.

**Review by TheSlimvReal — COMMENTED** [review]
Let's try to make more use of moments flexibility so we don't have to predefine all the units ourselves. Besides that looks good. Especially with the tests!

**Comment by christophscheuing** [comment]
The test for the indexOfCorrespondingDateRange is quite incomplete and inadequate, but apart from that this PR could probably soon be reviewed, @TheSlimvReal.

**Review by TheSlimvReal — APPROVED** [review]
Great work. The UI and functionality is very clean. I am looking forward to releasing this!

**Pull request merged**